pkg/R/DataFrame.R

```diff
@@ -774,21 +774,85 @@ setMethod("foreachPartition",
           })
 
 
-############################## DSL ##################################
+############################## SELECT ##################################
+
+getColumn <- function(x, c) {
+  column(callJMethod(x@sdf, "col", c))
+}
 
 setMethod("$", signature(x = "DataFrame"),
           function(x, name) {
-            column(callJMethod(x@sdf, "col", name))
+            getColumn(x, name)
+          })
+
+setMethod("$<-", signature(x = "DataFrame"),
+          function(x, name, value) {
+            stopifnot(class(value) == "Column")
```

Comment:
Fair point. That works for me, although I do still think the comment below (about being able to easily drop a column) is worth considering. It came up a lot when I was working through the SparkSQL demo.